### PR TITLE
Add support for pending transactions to `get_mempool_item_by_tx_id()` #9443

### PR DESCRIPTION
**Pull request by wallentx**
Co-Authored-By: @rwarren
Co-Authored-By: @emlowe 

Purpose: This PR is a clone of #9443 that had an unsigned commit

Current Behavior: https://github.com/Chia-Network/chia-blockchain/pull/9443 cannot be merged due to the first commit being unsigned

New Behavior: I took https://patch-diff.githubusercontent.com/raw/Chia-Network/chia-blockchain/pull/9443.diff and applied it on top of `main` in my branch, while also attributing credit to all who contributed in the original PR. There should be no drift in the intended effect of the original PR, and this one.

Breaking changes: None

### CODE REVIEW

File: chia/full_node/mempool_manager.py [hunk 1]
```diff
@@ -553,11 +553,17 @@ def get_spendbundle(self, bundle_hash: bytes32) -> Optional[SpendBundle]:
...
         return None
 
-    def get_mempool_item(self, bundle_hash: bytes32) -> Optional[MempoolItem]:
-        """Returns a MempoolItem if it's inside one the mempools"""
-        if bundle_hash in self.mempool.spends:
-            return self.mempool.spends[bundle_hash]
-        return None
+    def get_mempool_item(self, bundle_hash: bytes32, include_pending: bool = False) -> Optional[MempoolItem]:
+        """
+        Returns a MempoolItem if it's inside one the mempools.
```
Comment (arvidn) [inline chia/full_node/mempool_manager.py:558]: I can't parse this sentence. is it supposed to say "inside one *of* the mempools"?
But we only have one mempool, so I still wouldn't know what that means really. presumably "it" refers to the `bundle_hash` we pass in here, right?

**Review by arvidn — COMMENTED** [review]
looks correct to me (apart from the docstring). Do we have any documentation for our RPCs? If so, I would have expected to see an update to those in this PR as well.

**Comment by rwarren** [comment]
@arvidn  - That bit of the docstring is 100% unchanged from before my original [PR](https://github.com/Chia-Network/chia-blockchain/pull/9443/files) and, as you say, does make no sense.

When I authored the original PR here for the `include_pending` logic, I had explicitly chosen to not change the original docstring (aside from the bit about the new arg) since I didn't want to mix the new feature with a documentation fix, even if required.  Just stuck with what the PR was about.

**Review by arvidn — APPROVED** [review]

**Review by emlowe — APPROVED** [review]

File: chia/full_node/mempool_manager.py (hunk 1 above)
Comment (wallentx) [reply chia/full_node/mempool_manager.py:558]: I've spawned https://github.com/Chia-Network/chia-blockchain/issues/13984 for future cleanup to resolve this convo

**Pull request merged**